### PR TITLE
Make Kyma use knative-ingressgateway in 'knative' mode

### PR DESCRIPTION
**Pull request by sjanota**
**Description**
Based on #1803.
 
Changes proposed in this pull request:

- Kyma is configured to receive traffic through `knative-ingressgateway`
- There is switch in charts: `global.knative`. If set to true enables all of knative functionality. Without it kyma works as previously
- `--knative` adds overrides config map with `global.knative: true` to installer
- knative chart is installed on every setup, but chart is noop if knative is not enabled

**Related issue(s)**

Depends on: #1947
Depends on: #1948 
Depends on: #1949

### CODE REVIEW

File: resources/knative/templates/tests/test-serving.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{{ if .Values.global.knative }}
+  {{- $serviceName := .Values.global.knative | ternary "knative-ingressgateway" "istio-ingressgateway" -}}
```
Comment (jakkab) [inline resources/knative/templates/tests/test-serving.yaml]: Please remove duplicated conditions (.Values.global.knative)

File: resources/istio-kyma-patch/templates/job.yaml [hunk 2]
```diff
@@ -4,7 +4,6 @@ metadata:
   name: istio-kyma-patch
   annotations:
     "helm.sh/hook": post-install
-    "helm.sh/hook-delete-policy": hook-succeeded
```
Comment (jakkab) [inline resources/istio-kyma-patch/templates/job.yaml]: Please restore the annotation before merging!

File: resources/knative/README.md [hunk 3]
```diff
@@ -11,4 +11,5 @@ Included releases:
 Kyma-specific changes:
  * Every CRD has the `helm.sh/hook: crd-install` annotation set. This forces Helm to install the CRDs before other resources.
  * The duplicate of the `images.caching.internal.knative.dev` CRD is removed from the serving release.
- * If the **isLocalEnv** variable is set to `knative-ingressgateway`, the Service's type automatically changes to `NodePort`.
+ * `knative-ingressgateway` Service's `type` is changed to `NodePort`.
+ * `knative-shared-gateway` Gateway have TLS enabled
```
Comment (klaudiagrz) [inline resources/knative/README.md]:
```suggestion
 * `knative-shared-gateway` Gateway has TLS enabled
```

File: resources/knative/README.md [hunk 4]
```diff
@@ -11,4 +11,5 @@ Included releases:
 Kyma-specific changes:
  * Every CRD has the `helm.sh/hook: crd-install` annotation set. This forces Helm to install the CRDs before other resources.
  * The duplicate of the `images.caching.internal.knative.dev` CRD is removed from the serving release.
- * If the **isLocalEnv** variable is set to `knative-ingressgateway`, the Service's type automatically changes to `NodePort`.
+ * `knative-ingressgateway` Service's `type` is changed to `NodePort`.
```
Comment (klaudiagrz) [inline resources/knative/README.md]:
```suggestion
 * `knative-ingressgateway` Service **type** is changed to `NodePort`.
```

File: resources/knative/README.md [hunk 5]
```diff
@@ -11,4 +11,5 @@ Included releases:
 Kyma-specific changes:
  * Every CRD has the `helm.sh/hook: crd-install` annotation set. This forces Helm to install the CRDs before other resources.
  * The duplicate of the `images.caching.internal.knative.dev` CRD is removed from the serving release.
- * If the **isLocalEnv** variable is set to `knative-ingressgateway`, the Service's type automatically changes to `NodePort`.
+ * `knative-ingressgateway` Service **type** is changed to `NodePort`.
+ * `knative-shared-gateway` Gateway has TLS enabled
```
Comment (klaudiagrz) [inline resources/knative/README.md]:
```suggestion
 * `knative-shared-gateway` Gateway has TLS enabled.
```

**Pull request merged**